discovery/gossiper.go

```diff
@@ -1134,6 +1134,8 @@ func (d *AuthenticatedGossiper) networkHandler() {
 			for peerPub := range d.peerSyncers {
 				syncerPeers[peerPub] = struct{}{}
 			}
+
+			peerSyncers := d.peerSyncers
```

Comment:
Since `peerSyncers` is accessed w/o a lock below, it seems these values should be copied into a new map, as mutations in the original map will be reflected in `peerSyncers`

Reply:
Very true! Might as well just combine it with the above and use that below, will update.